nbconvert/nbconvertapp.py

```diff
@@ -30,6 +31,36 @@
 #Classes and functions
 #-----------------------------------------------------------------------------
 
+class CaselessStrEnumOrDottedObjectName(CaselessStrEnum, DottedObjectName):
```

Comment:
I'm not 100% sure of the simplicity/ugliness tradeof. Doing so also remove the automatic listing of known format from out `--help` (yes we can still do it manually) and move the check of wether an exporter format is right much later in the code. Plus that would duplicate quite a bit of logic already present. 

If you feel like `CaselessStrEnumOrDottedObjectName` is a too generic name, we can rename to `ExporterTrait` or something shorter. 

We can try though.